pkg/controller/release/release.go

```diff
@@ -42,6 +44,13 @@ import (
 	helmClient "github.com/crossplane-contrib/provider-helm/pkg/clients/helm"
 )
 
+const (
+	maxConcurrency = 10
+
+	resyncPeriod     = 10 * time.Minute
```

Comment:
Do you think this frequent of a resync period is needed? I'm guessing controllers already requeue appropriately when they need to periodically check

Reply:
It is a little hard to scale up with current default (1 minute, `longWait` in managed reconciler) since when the number of `Release` resources increases (imaging 1000 release's), controller will have them back into its queue before processing all and end up being blocked with infinite processing loop. 

Similarly, I believe this should also be a configurable option depending on the use case and trying to find some sensible defaults until we have ways to configure.

Reply:
Hmm is setting a more frequent resync going to address this issue? It seems like it could actually make it worse https://github.com/kubernetes-sigs/controller-runtime/issues/521

Reply:
I am actually setting managed reconciler `longWait` with [this](https://github.com/crossplane-contrib/provider-helm/pull/53/files#diff-7447a2eaf6d31130caf30f51cecffc5770f6538354ddae88487c5698d477357dR95). I named it `resyncPeriod` as I believe they doing the same thing. 
So I am trying to increase from 1 min to 10 min.

Reply:
Oh 😅 did not realize what file I was looking at and just assumed this was for the controller manager! Thanks for clarifying :)